server.go

```diff
@@ -0,0 +1,141 @@
+package srp
+
+import (
+	"bytes"
+	"crypto/subtle"
+	"encoding/base64"
+	"errors"
+	"math/big"
+
+	"github.com/ProtonMail/go-crypto/rand"
+)
+
+
+// Server stores the internal state for the validation of SRP proofs.
+type Server struct {
+	generator, modulus, verifier, serverSecret, serverEphemeral, sharedSession *big.Int
+	bitLength int
+}
+
+// NewServer creates a new server instance from the raw binary data.
+func NewServer(modulusBytes, verifier []byte, bitLength int) (*Server, error) {
+	var secret *big.Int
+	var err error
+
+	modulus := toInt(modulusBytes)
+	modulusMinusOne := big.NewInt(0).Sub(modulus, big.NewInt(1))
+
+	for {
+		secret, err = rand.Int(RandReader, modulusMinusOne)
+		if err != nil {
+			return nil, err
+		}
+
+		if secret.Cmp(big.NewInt(int64(bitLength*2))) > 0 { // Very likely
```

Comment:
Why is it checking this? (Is it meant to check that `secret` is at least `bitLength` bits long?)